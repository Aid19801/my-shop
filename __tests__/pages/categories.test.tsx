import { render, screen } from "@testing-library/react";
import Categories from "@/pages/categories";

jest.mock("../../lib/useUser", () => {
  return jest.fn(() => ({
    user: { id: 123 },
  }));
});

jest.mock("../../lib/useCategories", () => {
  return jest.fn(() => ({
    categories: ["Electronics", "jewelery"],
  }));
});

describe("CategoriesPage", () => {
  it("should render the page without exploding", () => {
    render(<Categories />);
    const heading = screen.getByText("Categories");
    expect(heading).toBeInTheDocument();
  });
  it("should match snapshot", () => {
    const { container } = render(<Categories />);
    expect(container).toMatchSnapshot();
  });
});
