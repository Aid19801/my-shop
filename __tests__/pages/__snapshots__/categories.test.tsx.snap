// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesPage should match snapshot 1`] = `
<div>
  <div
    class="basket__container"
  >
    <h4>
      Your Basket
    </h4>
  </div>
  <style>
    
        *,
        *::before,
        *::after {
          box-sizing: border-box;
        }
        body {
          margin: 0;
          color: #333;
          font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto,
            "Helvetica Neue", Arial, Noto Sans, sans-serif, "Apple Color Emoji",
            "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
        }
        .container {
          max-width: 65rem;
          margin: 1.5rem auto;
          padding-left: 1rem;
          padding-right: 1rem;
        }
      
  </style>
  <header>
    <nav>
      <ul>
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
      </ul>
    </nav>
    <style>
      
        ul {
          display: flex;
          list-style: none;
          margin-left: 0;
          padding-left: 0;
        }
        li {
          margin-right: 1rem;
          display: flex;
        }
        li:first-child {
          margin-left: auto;
        }
        a {
          color: #fff;
          text-decoration: none;
          display: flex;
          align-items: center;
        }
        a img {
          margin-right: 1em;
        }
        header {
          padding: 0.2rem;
          color: #fff;
          background-color: #333;
        }
      
    </style>
  </header>
  <main>
    <div
      class="container"
    >
      <h1>
        Categories
      </h1>
      <style>
        
      h1 {
        text-align: center;
        font-family: monospace;
        font-size: 50px;
        color: white;
        background: red;
        padding: 20px 20px;
        transform: skewY(-1deg);
      }
    
      </style>
      <div
        class="content__container"
      >
        <a
          class="card"
          href="/categories/Electronics"
        >
          <div
            class="card__infoContainer__top"
          >
            <h4>
              Electronics
            </h4>
          </div>
        </a>
        <style>
          
        .card {
          text-decoration: none;
          border: 2px solid rgba(0, 0, 0, 0.2);
          transition: 300ms ease-in-out;
          width: 300px;
          height: 200px;
          padding: 30px 50px;
          border-radius: 4px;
          box-shadow: rgb(0 0 0 / 20%) 0px 2px 1px -1px,
            rgb(0 0 0 / 14%) 0px 1px 1px 0px, rgb(0 0 0 / 12%) 0px 1px 3px 0px;
          background: rgba(255, 255, 255, 1);
          position: relative;
        }
        .card:hover {
          border: 2px solid red;
          background: rgba(255, 255, 255, 1);
        }
        .card__imgContainer {
          position: absolute;
          top: 0;
          left: 0;
          width: 100%;
          height: auto;
          max-height: 50%;
          min-height: 50%;
          contain: content;
        }
        .card__img {
          width: 100%;
          height: auto;
        }

        .card__infoContainer__bottom {
          padding: 10px;
          position: absolute;
          bottom: 0;
          left: 0;
          width: 100%;
          max-height: 50%;
          height: 50%;
          contain: content;
          position: absolute;
          height: 50%;
          display: flex;
          flex-wrap: wrap;
        }
        .card__infoContainer__top {
          height: 100%;
          display: flex;
          justify-content: center;
          align-items: center;
          text-align: center;
        }
        h4 {
          color: black;
          margin: 0;
        }
        p {
          color: grey;
          font-size: 9px;
        }
      
        </style>
        <a
          class="card"
          href="/categories/jewelery"
        >
          <div
            class="card__infoContainer__top"
          >
            <h4>
              jewelery
            </h4>
          </div>
        </a>
        <style>
          
        .card {
          text-decoration: none;
          border: 2px solid rgba(0, 0, 0, 0.2);
          transition: 300ms ease-in-out;
          width: 300px;
          height: 200px;
          padding: 30px 50px;
          border-radius: 4px;
          box-shadow: rgb(0 0 0 / 20%) 0px 2px 1px -1px,
            rgb(0 0 0 / 14%) 0px 1px 1px 0px, rgb(0 0 0 / 12%) 0px 1px 3px 0px;
          background: rgba(255, 255, 255, 1);
          position: relative;
        }
        .card:hover {
          border: 2px solid red;
          background: rgba(255, 255, 255, 1);
        }
        .card__imgContainer {
          position: absolute;
          top: 0;
          left: 0;
          width: 100%;
          height: auto;
          max-height: 50%;
          min-height: 50%;
          contain: content;
        }
        .card__img {
          width: 100%;
          height: auto;
        }

        .card__infoContainer__bottom {
          padding: 10px;
          position: absolute;
          bottom: 0;
          left: 0;
          width: 100%;
          max-height: 50%;
          height: 50%;
          contain: content;
          position: absolute;
          height: 50%;
          display: flex;
          flex-wrap: wrap;
        }
        .card__infoContainer__top {
          height: 100%;
          display: flex;
          justify-content: center;
          align-items: center;
          text-align: center;
        }
        h4 {
          color: black;
          margin: 0;
        }
        p {
          color: grey;
          font-size: 9px;
        }
      
        </style>
      </div>
      <style>
        
        .content__container {
          display: flex;
          flex-direction: row;
          flex-wrap: wrap;
          width: 100%;
          justify-content: space-around;
          align-items: stretch;
          gap: 30px;
        }
      
      </style>
    </div>
  </main>
  <style>
    
          .basket__container {
            position: fixed;
            z-index: 11;
            right: 10px;
            top: 150px;
            width: 180px;
            background: grey;
            border: 2px solid black;
            border-radius: 10px;
            padding: 20px;
          }
          .basket__item {
            border: 1px solid rgba(255, 255, 255, 0.2);
            background: lightgrey;
            padding: 5px;
            margin-bottom: 5px;
            transform: skewY(1deg);
            border-radius: 10px;
          }
          .basket__item p {
            color: black;
          }
          .basket__item h5 {
            color: black;
            margin: 0;
          }
          h4 {
            color: lightgrey;
            font-family: monospace;
          }
          p {
            color: white;
            font-size: 10px;
          }
        
  </style>
</div>
`;
